src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/BookingServiceTest.kt

```diff
@@ -2360,6 +2362,60 @@ class BookingServiceTest {
       }
     }
 
+
+    @Test
+    fun `createCancellation triggers emails when linked to Application`() {
```

Comment:
could we be a bit clearer maybe about the scenario here? I'm not clear on the circumstances where a booking will or won't have an application associated with it. Maybe we should could use this test to help clarify? 

Application seems to be optional in both "create" functions of the `BookingService`:

- `createApprovedPremisesAdHocBooking()`
 - `createApprovedPremisesBookingFromPlacementRequest()`

Reply:
This is another use case i don't fully understand. I'll add to my notes for tomorrow's meeting to see if we can get a better understanding of why this would happen and then improve how the tests describe this